### ISSUE TITLE
How to express if something is planned

### ISSUE THREAD

**Issue by TomMiksa** [RDA-DMP-Common/RDA-DMP-Common-Standard#46]
The application profile has a mechanism to [express that something is planned](https://github.com/RDA-DMP-Common/RDA-DMP-Common-Standard/blob/master/docs/FAQ.md#how-versioning-works)

However, there is a comment from Rob (@rwwh) who points out that: "if a trivial update is performed by a tool, and the modification date is changed, this can put it beyond the planned date for some action. Unless planned actions are indicated in an explicit way, this can be a problem! Every tool must now check that they don't inadvertently move the dmp modification date past any other date in the DMP!"

If we change our philosophy to make things more explicit then we likely need to come up with a controlled vocabulary such as <todo, doing, done, ??> to describe states. Then we need to introduce additional field to the dataset (where else?). 
The reason we did not do that in the first place:
- we were concerned that terms from a controlled vocabulary can be interpreted differently depending on the specific viewpoint, e.g. "done" can be understood by researcher as submitted to the funder, but for the funder it doesn't mean the action was approved
- we implemented the "dates arithmetic" for expressing embargoes in the same way as it was done for RIOXX - another application profile. Hence, it seemed a good idea to "talk about the future" in the same way. 

In my opinion, the suggestion made by @rwwh does not break the application profile, but shifts the responsibilities and is a change request that we may want to discuss and potentially address in later releases (maintenance). To my best knowledge, the existing (and pending) adoptions of the standard do not use the mechanism for expressing something is planed (did not get that far), so any changes in the specification should not break existing implementations.

**Comment by froggypaule** [comment]
I am afraid I don't understand the use case described above: I don't see the problem with how the chronology is implemented in [express that something is planned](https://github.com/RDA-DMP-Common/RDA-DMP-Common-Standard/blob/master/docs/FAQ.md#how-to-express-something-is-planned)...
What am I missing?

**Comment by rwwh** [comment]
We indicate that something is planned by having a date that is later than the release-date of the document. A change somewhere else changes the release date of the document, causing it to jump over the "planned by" date, changing its semantics to "this happened on that date"; however there is no guarantee that it actually took place.

**Comment by froggypaule** [comment]
Thanks @rwwh. So I suppose we do not want to rely on external (or internal?) checks to ensure that this situation does not happen ?
 Other than that, I have little insight into this issue... For now at least.

**Comment by rwwh** [comment]
I think the only sensible way to handle this is by using explicit semantics for any relevant dates.

I found: https://www.w3.org/TR/owl-time/ for semantically defining relations between time intervals, and http://semanticscience.org/resource/SIO_000033 "process status"

**Comment by froggypaule** [comment]
Hello.... yes, I agree: I had second thoughts, and now I understand why doing it this way will keep the CS self-contained and self-descriptive.

**Comment by paulwalk** [comment]
I agree with:

> I think the only sensible way to handle this is by using explicit semantics for any relevant dates.

Using the updated date of the DMP to mean anything other than "last time this DMP was updated" would be mistake.

**Comment by rduyme** [comment]
Based on discussions as part of OSTrails project. 
It seems that in the life cycle of a DMP, there are statements that  planned/promised ( ex: "I will use this FAIR repository X to store my datasets" and other that are statements of actions actually done during the project "This dataset as been created and is available in repository X, with doi Y". 

One use case of DMP is to write statements in a document, then a funder  gives you money from analysis of your statements, before the project has even started.

If one of the intent of DMP is also post evaluation of DMP statements. having this distinction is really relevant.  This would also mean there are statements of the DMP you are not supposed to update if an official evaluation was done on these.

**Comment by briri** [comment]
For the original problem statement, it seems to me that it's an issue with assumptions made about the date fields for specific use cases. I for one have been thinking that the "planned" date was in relation to the current date not the "modified" date on the DMP. This is also an assumption though! :) 

I think the modified date should reflect the last time the record was changed (in any way). To help address this particular scenario though, could we consider adding a `registered` or `published` date? I will be introducing this concept in the DMP Tool soon to denote when the DOI for the DMP is registered, but it need not be tied to DOI registration it could simply be used to indicate that the DMP is "complete" and was submitted to a funder or other stakeholder.

We should make sure we go through the documentation when we create v1.2 of the schema and be very explicit about the intended use of all dates

**Comment by MarekSuchanek** [comment]
This went through a discussion recently and the proposal is to keep it date-based as follows:

- Planned dataset = `dataset.issued is undefined || dataset.issued > current_timestamp >= dmp.modified`
- Produced dataset = `current_timestamp >= dmp.modified >= dataset.issued`
- Published / available dataset = is produced dataset and `len(dataset.distribution) >= 0` (has distributions, with access/download URL)
- Deleted dataset/distribution = `distribution.available_until <= current_timestamp`
- Reused dataset = #14 (using `dmp.created` is not suitable as you may reuse datasets created/published elsewhere over the course of this project/DMP)

Issue with `current_timestamp` is that as follows, e.g. for produced dataset:

- `dmp.modified >= dataset.issued` = at `dmp.modified` DMP stated that dataset has been already issued
- `current_timestamp >= dataset.issued > dmp.modified` = at `dmp.modified` DMP stated that dataset will be issued (and now it should be already issued); you can expect that the plan holds but it may not

 
As part of this issue, we can clarify this even more explicitly (wrt what is currently in FAQ). Does this cover all possible issues, or do we still miss something?

**Comment by andre-jernung** [comment]
I believe that this decision will create unnecessary limitations to the future usefulness of the standard for sharing of actual research project _planning_ between actors and tools, as opposed to just passively reporting what already happened.

> Use case: As a PI working at organisation _A_ during 2025, I might have an idea that I will need approximately _X_ TB of storage for a dataset (or let's say, _Y_ compute resources at a cluster) at organisation _B_ sometime during the second quarter of 2027.
> 
> If the only thing I could specify is that some future dataset will exist while `dataset.issued is undefined`, this will be largely useless information to share with organisation _B_.

To solve this I would need to be able to express that I am providing _approximations_ of dates set at in the future, expressed as f.e. a rough date interval or an approximated midpoint date. If this is available, in the use case organisation B would be able to provide cost calculations, do load balancing or procure new storage or compute resources to properly plan for what is going to happen sometime during the second quarter of 2027.

On the other hand I would of course also need to be able to update with an _exact_ date that I am sure of, which the standard already provides.

I think that to be truly useful for planning there is a need for some status flag to express ambiguity of dates, perhaps even graded such as: `date_certainty: exact | approximated | estimated`

**Comment by TomMiksa** [comment]
If I think of a `Distribution` class, I can imagine a case where some aspects (like `format` or `size`) are still unknown, while other attributes (such as `host` or `license`) are already fixed.  

If we assign `certainty` at the `Distribution` level, then all contained information — even the fixed parts — would also be marked as uncertain.  

**Is that desirable or not?**

Scenarios to Consider

1. **Repository updates from uncertain DMP data**  
   Suppose a repository receives a DMP that includes a `Distribution` marked as uncertain (as described above).  
   - If the repository wants to set a `license` based on that `Distribution`, should it proceed?  
   - Or should it ignore the data because the entire `Distribution` is uncertain?

2. **Inheritance of certainty**  
   If a `Distribution` is marked as uncertain, does that automatically make its linked information (like `Host`) uncertain as well?  
   - In other words, does `certainty` propagate (inherit) to related entities?


Alternative: Field-Level Certainty

We could instead apply `certainty` at the **field level**.  
However, that would effectively double the number of fields, since each would need an accompanying certainty flag.  
While these flags could be optional, it feels like a significant increase in complexity and potential overhead.

Summary:
- Should `certainty` apply to the entire `Distribution`, or individually to its fields?  
- How should inheritance or propagation of `certainty` be handled?
